### ISSUE TITLE
Crashed when restart webp animated drawable on Android4.x

### ISSUE THREAD

**Issue by kifile** [facebook/fresco#1434]
I place a draweeview in my layout, and display an animated webp on it.

I stop the animation manually after staring the webp animation,. Sometimes  when the user click the widget to replay the webp, it crashed.

The stacktrace is as follow:

```
Fatal Exception: java.lang.Error: signal 11 (Address not mapped to object) at address 0x0 [at libc.so:0x16618 (getauxval+0x8)]
       at com.facebook.animated.webp.WebPFrame.nativeRenderFrame(Proguard)
       at com.facebook.animated.webp.WebPFrame.renderFrame(Proguard:52)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableBackendImpl.renderImageSupportsScaling(Proguard:205)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableBackendImpl.renderFrame(Proguard:179)
       at com.facebook.imagepipeline.animated.impl.AnimatedImageCompositor.renderFrame(Proguard:114)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableCachingBackendImpl.runPrefetch(Proguard:454)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableCachingBackendImpl.access$200(Proguard:53)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableCachingBackendImpl$3.call(Proguard:412)
       at bolts.Task$4.run(Proguard:357)
       at com.facebook.common.executors.ConstrainedExecutorService$Worker.run(Proguard:178)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
       at com.facebook.imagepipeline.core.PriorityThreadFactory$1.run(Proguard:43)
       at java.lang.Thread.run(Thread.java:841)
Caused by java.lang.Error: signal 11 (Address not mapped to object) at address 0x0 [at libc.so:0x16618 (getauxval+0x8)]
       at com.facebook.animated.webp.WebPFrame.nativeRenderFrame(Proguard)
       at com.facebook.animated.webp.WebPFrame.renderFrame(Proguard:52)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableBackendImpl.renderImageSupportsScaling(Proguard:205)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableBackendImpl.renderFrame(Proguard:179)
       at com.facebook.imagepipeline.animated.impl.AnimatedImageCompositor.renderFrame(Proguard:114)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableCachingBackendImpl.runPrefetch(Proguard:454)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableCachingBackendImpl.access$200(Proguard:53)
       at com.facebook.imagepipeline.animated.impl.AnimatedDrawableCachingBackendImpl$3.call(Proguard:412)
       at bolts.Task$4.run(Proguard:357)
       at com.facebook.common.executors.ConstrainedExecutorService$Worker.run(Proguard:178)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
       at com.facebook.imagepipeline.core.PriorityThreadFactory$1.run(Proguard:43)
       at java.lang.Thread.run(Thread.java:841)
```

I find the crash only appears in Android4.x.  Whether this is related to the image memory strategy.

**Comment by oprisnik** [comment]
Do you have a sample image / code / minimal example so that we can reproduce the issue?

**Comment by kifile** [comment]
Here is my code. I use a handler to control the animated drawable. But sometimes the crash appears on Android 4.x, I only see it in Fabric and cannot produce it.
```
public class QuestionDialogFragment extends DialogFragment
        implements View.OnClickListener {

    private Handler mHandler;

    public static class GuideHandler extends LeakGuardHandlerWrapper<QuestionDialogFragment> {

        private static final int MSG_STOP = 1;
        private static final int MSG_START = 2;

        public GuideHandler(QuestionDialogFragment ownerInstance) {
            super(ownerInstance);
        }

        @Override
        public void handleMessage(Message msg) {
            super.handleMessage(msg);
            QuestionDialogFragment fragment = getOwnerInstance();
            if (fragment == null) {
                return;
            }
            switch (msg.what) {
                case MSG_START: {
                    SimpleDraweeView draweeView = fragment.mDraweeView;
                    if (draweeView != null) {
                        DraweeController controller = draweeView.getController();
                        if (controller != null) {
                            Animatable animatable = controller.getAnimatable();
                            if (animatable != null) {
                                if (animatable instanceof AnimatedDrawable) {
                                    try {
                                        Field field = AbstractAnimatedDrawable.class
                                                .getDeclaredField("mDurationMs");
                                        field.setAccessible(true);
                                        int time = (Integer) field.get(animatable);
                                        sendMessageDelayed(obtainMessage(MSG_STOP), time);
                                    } catch (Exception ignore) {
                                        // Ignore reflection exception.
                                    }
                                }
                                animatable.start();
                                if (fragment.mPlayView != null) {
                                    fragment.mPlayView.setVisibility(View.GONE);
                                }
                            }
                        }
                    }
                    break;
                }
                case MSG_STOP: {
                    SimpleDraweeView draweeView = fragment.mDraweeView;
                    if (draweeView != null) {
                        DraweeController controller = draweeView.getController();
                        if (controller != null) {
                            Animatable animatable = controller.getAnimatable();
                            if (animatable != null) {
                                animatable.stop();
                            }
                        }
                    }
                    if (fragment.mPlayView != null) {
                        fragment.mPlayView.setVisibility(View.VISIBLE);
                    }
                    break;
                }
                default:
                    break;
            }

        }
    }

    private SimpleDraweeView mDraweeView;
    private View mPlayView;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        setCancelable(false);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_dialog_question, container, false);
        final Resources resources = getResources();
        int width = resources.getDisplayMetrics().widthPixels;
        int padding = resources.getDimensionPixelOffset(R.dimen.dialog_default_margin);
        width = Math.min(width - 2 * padding,
                resources.getDimensionPixelOffset(R.dimen.dialog_max_width));
        View dialogContainer = view.findViewById(R.id.dialog_container);
        ViewGroup.LayoutParams params = dialogContainer.getLayoutParams();
        params.width = width;
        dialogContainer.setLayoutParams(params);
        return view;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        view.findViewById(R.id.close).setOnClickListener(this);
        view.findViewById(R.id.guide_go).setOnClickListener(this);
        mHandler = new GuideHandler(this);
        mDraweeView = (SimpleDraweeView) view.findViewById(R.id.guide_animation);
        mPlayView = view.findViewById(R.id.play);
        mPlayView.setOnClickListener(this);
        ImageRequest request = ImageRequestBuilder
                .newBuilderWithSource(Uri.parse("res:///" + R.raw.guide_animation))
                .setImageDecodeOptions(
                        ImageDecodeOptions.newBuilder().setDecodePreviewFrame(true).build())
                .setAutoRotateEnabled(true).build();
        DraweeController controller = Fresco.newDraweeControllerBuilder().setImageRequest(request)
                .setOldController(mDraweeView.getController()).setAutoPlayAnimations(false).build();
        mDraweeView.setController(controller);
        mHandler.sendEmptyMessage(GuideHandler.MSG_STOP);
    }

    @Override
    public void onDestroy() {
        if (mHandler != null) {
            mHandler.removeCallbacksAndMessages(null);
            mHandler = null;
        }
        super.onDestroy();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.close: {
                dismissAllowingStateLoss();
                break;
            }
            case R.id.guide_go: {
                dismissAllowingStateLoss();
                break;
            }
            case R.id.play: {
                play();
                break;
            }
            default:
                break;
        }
    }

    private void play() {
        mHandler.sendEmptyMessage(GuideHandler.MSG_START);
    }

}
```

**Comment by kifile** [comment]
Before I using handler, I call setController to play a same animated webp image each time when play() method is called, but I saw the same crash on Fabric.

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated GIFs and animated WebPs that is way more performant, especially in the case where you display multiple images. Let us know if you're still experiencing problems with 1.4.0.
